code/game/machinery/station_map.dm

```diff
@@ -0,0 +1,321 @@
+			else
+				to_chat(user, "<span class='notice'>An hologram of the station appears before your eyes.</span>")
+
+/obj/machinery/station_map/attack_paw(var/mob/user)
+	src.attack_hand(user)
+
+/obj/machinery/station_map/attack_animal(var/mob/user)
+	src.attack_hand(user)
+
+/obj/machinery/station_map/attack_ai(var/mob/user)
+	return//TODO: Give AIs their own holomap
+
+/obj/machinery/station_map/process()
+	if((stat & (NOPOWER|BROKEN)) || !anchored)
+		stopWatching()
+
+	checkPosition()
+
+/obj/machinery/station_map/proc/checkPosition()
+	if(!watching_mob || (watching_mob.loc != loc) || (dir != watching_mob.dir))
+		stopWatching()
+
+/obj/machinery/station_map/proc/stopWatching()
+	if(watching_mob)
+		if(watching_mob.client)
+			var/mob/M = watching_mob
+			spawn(5)//we give it time to fade out
+				M.client.images -= station_map
+		watching_mob.callOnFace -= "\ref[src]"
+	watching_mob = null
+	animate(station_map, alpha = 0, time = 5, easing = LINEAR_EASING)
+
+/obj/machinery/station_map/power_change()
+	. = ..()
+	update_icon()
+
+/obj/machinery/station_map/proc/set_broken()
+	stat |= BROKEN
+	update_icon()
+
+/obj/machinery/station_map/update_icon()
+	overlays.len = 0
+	if(stat & BROKEN)
+		icon_state = "station_mapb"
+	else if((stat & NOPOWER) || !anchored)
+		icon_state = "station_map0"
+	else
+		icon_state = "station_map"
+
+		if(bogus)
+			station_map.overlays.len = 0
+			station_map.overlays |= legend
+		else
+			switch(dir)
+				if(NORTH)
+					small_station_map.icon = extraMiniMaps[HOLOMAP_EXTRA_STATIONMAPSMALL_NORTH+"_[original_zLevel]"]
+				if(SOUTH)
+					small_station_map.icon = extraMiniMaps[HOLOMAP_EXTRA_STATIONMAPSMALL_SOUTH+"_[original_zLevel]"]
+				if(EAST)
+					small_station_map.icon = extraMiniMaps[HOLOMAP_EXTRA_STATIONMAPSMALL_EAST+"_[original_zLevel]"]
+				if(WEST)
+					small_station_map.icon = extraMiniMaps[HOLOMAP_EXTRA_STATIONMAPSMALL_WEST+"_[original_zLevel]"]
+
+			overlays |= small_station_map
+			station_map.overlays.len = 0
+			if(map.holomap_offset_x.len >= original_zLevel)
+				cursor.pixel_x = (x-6+map.holomap_offset_x[original_zLevel])*PIXEL_MULTIPLIER
+				cursor.pixel_y = (y-6+map.holomap_offset_y[original_zLevel])*PIXEL_MULTIPLIER
+			else
+				cursor.pixel_x = (x-6)*PIXEL_MULTIPLIER
+				cursor.pixel_y = (y-6)*PIXEL_MULTIPLIER
+			station_map.overlays |= cursor
+			station_map.overlays |= legend
+
+	switch(dir)
+		if(NORTH)
+			pixel_x = 0
+			pixel_y = WORLD_ICON_SIZE
+		if(SOUTH)
+			pixel_x = 0
+			pixel_y = -1*WORLD_ICON_SIZE
+		if(EAST)
+			pixel_x = WORLD_ICON_SIZE
+			pixel_y = 0
+		if(WEST)
+			pixel_x = -1*WORLD_ICON_SIZE
+			pixel_y = 0
+
+	if(floor_markings)
+		floor_markings.dir = dir
+		floor_markings.pixel_x = -1*pixel_x
+		floor_markings.pixel_y = -1*pixel_y
+		overlays |= floor_markings
+
+	if(panel_open)
+		overlays |= "station_map-panel"
+	else
+		overlays -= "station_map-panel"
+
+/obj/machinery/station_map/ex_act(severity)
+	switch(severity)
+		if(1.0)
```

Comment:
Redundant decimals, remove the `.0`.

They always just originated from a bug in the decompiler used to decompile SS13 for the first time, I heard.